File: src/test/java/com/dotnomi/fabricdependencyinjection/java/testclasses/success/ServiceA.java
package com.dotnomi.fabricdependencyinjection.java.testclasses.success;

import com.dotnomi.fabricdependencyinjection.annotation.ModInject;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
public class ServiceA {
  private final ServiceB serviceB;

  @ModInject
  public ServiceA(ServiceB serviceB) {
    this.serviceB = serviceB;
  }

  public ServiceB getServiceB() {
    return serviceB;
  }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/success/ServiceA.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.success

import com.dotnomi.fabricdependencyinjection.annotation.ModInject
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
class ServiceA @ModInject constructor(val serviceB: ServiceB)
